torax/_src/pedestal_model/tests/register_model_test.py

```diff
@@ -0,0 +1,175 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for pedestal model registration."""
+
+from typing import Annotated, Literal
+from absl.testing import absltest
+import chex
+from torax._src import geometry
+from torax._src import state
+from torax._src.pedestal_model import pedestal_model
+from torax._src.pedestal_model import pydantic_model
+from torax._src.pedestal_model import register_model
+from torax._src.pedestal_model import runtime_params as pedestal_runtime_params
+from torax._src.torax_pydantic import torax_pydantic
+import jax.numpy as jnp
+
+
+class RegisterModelTest(absltest.TestCase):
+  """Tests for pedestal model registration functionality."""
+
+  def test_register_custom_pedestal_model(self):
+    """Test that a custom pedestal model can be registered and used."""
+
+    # Define a simple custom pedestal model
+    @chex.dataclass(frozen=True)
+    class TestPedestalModel(pedestal_model.PedestalModel):
+      """Test pedestal model."""
+
+      def _call_implementation(
+          self,
+          runtime_params: 'TestRuntimeParams',
+          geo: geometry.Geometry,
+          core_profiles: state.CoreProfiles,
+      ) -> pedestal_model.PedestalModelOutput:
+        """Return fixed test values."""
+        rho_norm_ped_top = 0.95
+        rho_norm_ped_top_idx = jnp.argmin(
+            jnp.abs(geo.rho_norm - rho_norm_ped_top)
+        )
+
+        return pedestal_model.PedestalModelOutput(
+            rho_norm_ped_top=rho_norm_ped_top,
+            rho_norm_ped_top_idx=rho_norm_ped_top_idx,
+            T_i_ped=runtime_params.test_value,
+            T_e_ped=runtime_params.test_value,
+            n_e_ped=runtime_params.test_value * 1e20,
+        )
+
+    @chex.dataclass(frozen=True)
+    class TestRuntimeParams(pedestal_runtime_params.RuntimeParams):
+      """Test runtime parameters."""
+      test_value: float = 42.0
+
+    class TestPedestal(pydantic_model.BasePedestal):
+      """Test Pydantic config."""
+
+      model_name: Annotated[Literal['test_pedestal'], torax_pydantic.JAX_STATIC] = (
+          'test_pedestal'
+      )
+      test_value: float = 42.0
+
+      def build_pedestal_model(self) -> TestPedestalModel:
+        return TestPedestalModel()
+
+      def build_runtime_params(
+          self, t: chex.Numeric
+      ) -> TestRuntimeParams:
+        return TestRuntimeParams(
+            set_pedestal=self.set_pedestal.get_value(t),
+            test_value=self.test_value,
+        )
+
+    # Register the model
+    register_model.register_pedestal_model(TestPedestal)
+
+    # Verify it can be instantiated
+    config = TestPedestal(test_value=99.0)
```

Comment:
What we need to test is that it can be instantiated through the ModelConfig API not directly which works even without the test